### PR TITLE
Reset TPM on boot

### PR DESCRIPTION
**Pull request by Ham22**
Added mfio support to be able to reset the TPM via gpio. Additionally found that board_late_init wasn't being called so enabled it and moved watchdog support out into it's standard location.

Tested by making sure that both a modified tpm reset board and a current version board are able to boot with a working tpm and still work after being reset (using reset button, using software command and cycling power).

This connects CreatorDev/openwrt#32

### CODE REVIEW

File: board/imgtec/pistachio_bub/mfio.h [hunk 1]
```diff
@@ -189,3 +189,7 @@ void mfio_setup_usb_pwr(void);
 #if defined(CONFIG_TARGET_PISTACHIO_MARDUK)
 void mfio_setup_led(void);
 #endif
+
+#if defined(CONFIG_TPM)
```
Comment (abhijit-mahajani-imgtec) [inline board/imgtec/pistachio_bub/mfio.h]: why not CONFIG_TARGET_PISTACHIO_MARDUK?

**Review by abhijit-mahajani-imgtec — CHANGES_REQUESTED** [review]

File: board/imgtec/pistachio_bub/mfio.h (hunk 1 above)
Comment (Ham22) [inline board/imgtec/pistachio_bub/mfio.h]: It doesn't exist here

Comment (Ham22) [reply board/imgtec/pistachio_bub/mfio.h]: Done

**Review by abhijit-mahajani-imgtec — APPROVED** [review]
Verified with old bootloader not giving error on powered up
tpm init && tpm startup TPM_ST_CLEAR
but now its giving error since already initialized.

pistachio # tpm init && tpm startup TPM_ST_CLEAR
Error: -16

**Pull request merged**